src/index.js

```diff
@@ -16,14 +16,23 @@ class Dropzone extends React.Component {
     this.onDragLeave = this.onDragLeave.bind(this);
     this.onDragOver = this.onDragOver.bind(this);
     this.onDrop = this.onDrop.bind(this);
+    this.onFileDialogCancel = this.onFileDialogCancel.bind(this);
 
     this.state = {
-      isDragActive: false
+      isDragActive: false,
+      isFileDialogActive: false
     };
   }
 
   componentDidMount() {
     this.enterCounter = 0;
+    // Tried implementing addEventListener, but didn't work out
+    document.body.onfocus = this.onFileDialogCancel;
```

Comment:
That's weird. I would assume it can be the React that is messing around with the event listeners here... I could investigate further.